htlcswitch/spider_test.go

```diff
@@ -6,364 +6,216 @@ import (
+	// the overflowQueue was working properly.
+	firstPaymentErr := <-c
+	if firstPaymentErr != nil {
+		fmt.Println(firstPaymentErr)
+		t.Fatal("error has been received in first payment, alice->bob")
+	}
 }
 
 // bob<->alice channel has insufficient BTC capacity/bandwidth. In this test we
 // send the payment from Alice to Carol over the Bob peer. (Alice -> Bob ->
-// Carol) with the Bob -> Carol link being the bottleneck.
-// Here, Carol does not send Bob any money, and this payment should eventually
-// timeout.
-func TestSpiderInsufficentFunds(t *testing.T) {
+// Carol) with the Bob -> Carol link being the bottleneck.  Here, Carol does
+// not send Bob enough money, and this payment should eventually timeout.
+func TestSpiderInsufficientFunds(t *testing.T) {
 	t.Parallel()
-
-	channels, cleanUp, _, err := createClusterChannels(
-		btcutil.SatoshiPerBitcoin*5,
-		btcutil.SatoshiPerBitcoin*3)
-		if err != nil {
-			t.Fatalf("unable to create channel: %v", err)
-		}
-		defer cleanUp()
-
-		n := newThreeHopNetwork(t, channels.aliceToBob, channels.bobToAlice,
-		channels.bobToCarol, channels.carolToBob, testStartingHeight)
-		if err := n.start(); err != nil {
-			t.Fatalf("unable to start three hop network: %v", err)
-		}
-		defer n.stop()
-
-		amount := lnwire.NewMSatFromSatoshis(4 * btcutil.SatoshiPerBitcoin)
-		htlcAmt, totalTimelock, hops := generateHops(amount, testStartingHeight,
-					n.firstBobChannelLink, n.carolChannelLink)
-
-		// What we expect to happen:
-		// * HTLC add request to be sent to from Alice to Bob.
-		// * Alice<->Bob commitment states to be updated.
-		// * Bob trying to add HTLC add request in Bob<->Carol channel.
-		// * Not enough funds for that, so it gets added to overFlowQueue in
-		// Bob->Carol channel.
-		// Payment times out.
-
-		firstHop := n.firstBobChannelLink.ShortChanID()
-		// launch second payment here, which sleeps for a bit, and then pays from
-		// carol to bob.
-		_, err = n.makePayment(
-			n.aliceServer, n.carolServer, firstHop, hops, amount, htlcAmt,
-			totalTimelock,
-		).Wait(10 * time.Second)
-		// modifications on existing test. We do not want it to fail here.
-		if err == nil {
-			fmt.Println(err)
-			t.Fatal("no error was received in the payment, alice->bob")
-		}
-		fmt.Println("alice -> carol failed succesfully")
-		// if we reach this point, then all the payments have succeeded.
+	n, cleanUp := StartThreeHopNetwork(5, 3, t)
+	defer cleanUp()
+	defer n.stop()
+	// First, we will send money from Alice -> Carol. This should block on Bob ->
+	// Carol due to insufficient funds.
+	// What we expect to happen:
+	// * HTLC add request to be sent to from Alice to Bob.
+	// * Alice<->Bob commitment states to be updated.
+	// * Bob trying to add HTLC add request in Bob<->Carol channel.
+	// * Not enough funds for that, so it gets added to overFlowQueue in
+	// Bob->Carol channel.
+	// * Carol sends Bob money.
+	// * Bob -> Carol channel now has enough money, and the Bob -> Carol
```

Comment:
Think this comment needs to be edited, but apart from that this flow is significantly clearer!